File: reverse/reverse_int.rb
# 反转整数，但是正负号不变
def reverse_int x
  sign = x > 0 ? 1 : -1
  # Abs是absolute,也就是绝对值
  x = x.abs
  # 反转整数
  reversed = 0
  while x >0
      reversed = reversed * 10 + x % 10
      x /= 10
  end
  reversed *= sign
  return 0 if reversed < -2 ** 31 || reversed > 2 ** 31 -1
  reversed
end
x = 1342
y = -1342
puts reverse_int x
puts reverse_int y

File: reverse/reverse_int.go
package main

import (
  "fmt"
  "math"
)

func reverseInt(x int) int {
  sign := 1
  if x < 0 {
    sign = -1
    x = -x
  }
  // 反转整数
  reversed := 0
  for x > 0 {
    reversed = reversed *10 + x %10
    x /= 10
  }
  reversed *= sign
  if reversed < math.MinInt32 || reversed > math.MaxInt32 {
    return 0
  }
  return reversed
}

func main() {
  x := -1234
  y := 1234
  reversed_x := reverseInt(x)
  reversed_y := reverseInt(y)
  fmt.Println(reversed_x)
  fmt.Println(reversed_y)

}
